scripts/python/amend_user_repository_access.py

```diff
@@ -0,0 +1,478 @@
+                        team_name, repository_name)
+                    # Depends who adds the oauth_token to repo is added to every team
+                    github_service.remove_user_from_team(
+                        "AntonyBishop", team_id)
+                    github_service.remove_user_from_team("nickwalt01", team_id)
+                    github_service.remove_user_from_team("ben-al", team_id)
+                    github_service.remove_user_from_team(
+                        "moj-operations-engineering-bot", team_id)
+
+                github_service.amend_team_permissions_for_repository(
+                    team_id, users_permission, repository_name)
+
+                github_service.add_user_to_team(username, team_id)
+                github_service.remove_user_from_repository(
+                    username, repository_name)
+            except Exception:
+                logging.exception(
+                    f"Exception while putting {username} into team. Skipping iteration.")
+
+
+def run(github_service: GithubService, badly_named_repositories: list[str], repo_issues_enabled):
+    """A function for the main functionality of the script"""
+
+    # Get the usernames of the outside collaborators
+    outside_collaborators = github_service.get_outside_collaborators_login_names()
+
+    # Get the MoJ organisation teams and users info
+    org_teams = fetch_teams(github_service)
+
+    # Get the MoJ organisation repos and direct users
+    org_repositories = fetch_repositories(
+        github_service, outside_collaborators, repo_issues_enabled)
+
+    # loop through each organisation repository
+    for repository in org_repositories:
+
+        if repository.name not in badly_named_repositories:
+            # close any previously opened issues that have expired
+            github_service.close_expired_issues(repository.name)
+
+            users_not_in_a_team = repository.direct_members
+
+            remove_users_with_duplicate_access(github_service, repo_issues_enabled,
+                                               repository.name,
+                                               repository.direct_members,
+                                               users_not_in_a_team,
+                                               org_teams,
+                                               )
+
+            remaining_users = users_not_in_a_team
+
+            for username in users_not_in_a_team:
+                put_user_into_existing_team(github_service,
+                                            repository.name, username, remaining_users, org_teams
+                                            )
+
+            put_users_into_new_team(
+                github_service, repository.name, remaining_users)
+
+
+def main():
+    if len(sys.argv) == 2:
+        oauth_token = sys.argv[1]
+    else:
+        raise ValueError("Missing a script input parameter")
+    github_service = GithubService(oauth_token, "ministryofjustice")
```

Comment:
Kept the organisation name

Reply:
Refactoring it out later on?

Reply:
That's the plan 🌱 😁 

I'll create a separate ticket for that 🎫